src/mudlet.cpp

```diff
@@ -1315,8 +1316,10 @@ void mudlet::closeHost(const QString& name)
     // instead we just have to accept that any profile changes will not be
     // saved if the preferences dialog is not closed before the profile is...
     int hostCount = mHostManager.getHostCount();
-    emit signal_hostDestroyed(pH, --hostCount);
     mHostManager.deleteHost(pH->getName());
+    // pH is NOT valid now - it must NOT be dereferenced:
+    emit signal_hostDestroyed(pH, --hostCount);
```

Comment:
Isn't that what this line is doing though?

Reply:
Well, it is signalling to *anything* connected to it that that pointer is no longer valid - but as it has been moved to *after* the point where the destruction happens (in the `HostManage::deleteHost(...)` call) it no longer signals just the *intent* to destroy the relevant profile but that it has **actually happened**.

Reply:
Hmm I don't mind the change, but that's quite an important one difference in how things work now - how is that related to this PR?

Reply:
It is needed so that the changes to the accessible names/descriptions have the right count of profiles. :man_facepalming: - which is the second argument in the signal - which I didn't spot when I wired up the connection to the `(void) TConsole::slot_adjustAccessibleNames()` - ***because the newfangled (compile time checking) invocations of `QObject::connect(...)` no longer mentions the arguments and only shows the function pointers***...

Reply:
Ah, got that wrong - it was the `(void) mudlet::signal_hostCreated(Host*, const quint8)` and `(void) mudlet::signal_hostDestroyed(Host*, const quint8)` signals that bore that information...

Reply:
Lets have it as a separate PR so we can verify the code is OK and we can test its consequences properly. I agree it's a dependency, and having it separately will make it easier to test.

Reply:
So I should just put the existing line back in it's original place.

Reply:
Original line now put back as it was.